Loop/Managers/DeviceDataManager.swift

```diff
@@ -1047,16 +1049,15 @@ extension DeviceDataManager: PumpManagerDelegate {
         dispatchPrecondition(condition: .onQueue(.main))
         log.default("PumpManager:%{public}@ hasNewPumpEvents (lastReconciliation = %{public}@)", String(describing: type(of: pumpManager)), String(describing: lastReconciliation))
 
-        doseStore.addPumpEvents(events, lastReconciliation: lastReconciliation, replacePendingEvents: replacePendingEvents) { (error) in
-            if let error = error {
+        Task {
+            do {
+                try await doseStore.addPumpEvents(events, lastReconciliation: lastReconciliation, replacePendingEvents: replacePendingEvents)
+            } catch {
                 self.log.error("Failed to addPumpEvents to DoseStore: %{public}@", String(describing: error))
+                completion(error)
             }
-
-            completion(error)
-
-            if error == nil {
-                NotificationCenter.default.post(name: .PumpEventsAdded, object: self, userInfo: nil)
-            }
+            completion(nil)
```

Comment:
won't this call the completion twice where there is an error? Is there a `return` missing from the catch?